petabvis/window_functionality.py

```diff
@@ -176,25 +180,43 @@ def table_tree_view(window: QtWidgets.QMainWindow, folder_path):
         branch.appendRow(file)
         root_node.appendRow(branch)
 
-    if window.simulation_df is not None:
-        branch = QtGui.QStandardItem(C.SIMULATION_TABLES)
-        simulation_file = QtGui.QStandardItem(C.SIMULATION_FILE)
-        df = window.simulation_df
-        simulation_file.setData({"df": df,
-                                 "name": C.SIMULATION_FILE},
-                                role=C.USER_ROLE)
-        branch.appendRow(simulation_file)
-        root_node.appendRow(branch)
-
     tree_view.setModel(model)
     tree_view.expandAll()
+    window.tree_root_node = root_node
     reconnect(tree_view.clicked,
               lambda i: exchange_dataframe_on_click(i, model,
                                                     window, tidy_names))
     reconnect(tree_view.doubleClicked,
               lambda i: display_table_on_doubleclick(i, model, window))
 
 
+def add_simulation_df_to_tree_view(window, filename):
+    """
+    Add the simulation df to the tree_view of the main window.
+    If no simulation df has been added to the tree_view yet,
+    also create a "Simulation Tables" branch.
+
+    Arguments:
+        window: Main window to add the simulation df to.
+        filename: The name of the simulation file.
+    """
+    simulation_df = window.simulation_df
+    root_node = window.tree_root_node
+    simulation_branch = window.simulation_tree_branch
+    if simulation_branch is None:
+        simulation_branch = QtGui.QStandardItem(C.SIMULATION_TABLES)
+        simulation_branch.setEditable(False)
+        root_node.appendRow(simulation_branch)
+        window.simulation_tree_branch = simulation_branch
+
+    simulation_file = QtGui.QStandardItem(filename)
+    simulation_file.setData({"df": simulation_df,
+                             "name": filename},
```

Comment:
`"df"` and `"name"` could be constants (and then the code edited to use these constants everywhere)